contrib/avro/tests/java/org/pantsbuild/contrib/avro/BUILD

```diff
@@ -0,0 +1,12 @@
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+junit_tests(
+  dependencies=[
+    '3rdparty:junit',
+    'contrib/avro/tests/avro/org/pantsbuild/contrib/avro:simple',
+    'contrib/avro/tests/avro/org/pantsbuild/contrib/avro:user',
+    # FIXME(tdyas): Needed because synthetic target is not getting injected deps?
+    '//:avro-java-runtime',
```

Comment:
I'm guessing that this is a strict_deps issue. Have you tried add a declaration for `def synthetic_target_extra_exports(self, target, target_workdir):` that resolves as '//:avro-java-runtime'?

You could test whether that's the issue by adding the kwarg `strict_deps=False` to this target and removing the explicit dependency here.

https://www.pantsbuild.org/3rdparty_jvm.html#strict-dependencies